Code/Editor/MainWindow.cpp

```diff
@@ -845,78 +853,189 @@ void MainWindow::InitActions()
         .SetShortcut(tr("Z"))
         .SetToolTip(tr("Center on Selection (Z)"))
         .Connect(&QAction::triggered, this, &MainWindow::OnGotoSelected);
+
+    const auto goToViewBookmarkFn = [](int index)
+    {
+        AzToolsFramework::ViewBookmarkLoaderInterface* bookmarkLoader = AZ::Interface<ViewBookmarkLoaderInterface>::Get();
+        if (!bookmarkLoader)
+        {
+            AZ_Warning("Main Window", false, "Couldn't find View Bookmark Loader");
+            return false;
+        }
+
+        const AZStd::optional<AzToolsFramework::ViewBookmark> bookmark =
+            bookmarkLoader->LoadBookmarkAtIndex(index);
+
+        if (!bookmark.has_value())
+        {
+            return false;
+        }
+
+        // Check the bookmark we want to load is not exactly 0 
+        if (bookmark.value() == ViewBookmark())
```

Comment:
This reeks of code smell when there is a comparison against a default constructed entry of a class.
Is there is no IsValid() function on the `ViewBookmark` class?

Reply:
Yes, I agree, just changed it!